compiler/rustc_lint_defs/src/builtin.rs

```diff
@@ -2592,8 +2592,8 @@ declare_lint! {
     ///
     /// The fix to this is to wrap the unsafe code in an `unsafe` block.
     ///
-    /// This lint is "allow" by default since this will affect a large amount
-    /// of existing code, and the exact plan for increasing the severity is
+    /// This lint is "allow" by default on editions up to 2021, from 2024 it is
+    /// "warn" by default, the plan for increasing severity after this is
```

Comment:
It was my impression from the FCP discussion that there is still want to make this `deny`-by-default or a hard-error in the future, so I have left this mention of increasing the severity in.

Reply:
We discussed that particular point in lang triage today, and agree with "still being considered".

I think all of "stays warning", "becomes deny-by-default lint", and "hard error" are all on the table still, pending experience we get with doing this in 2024.